.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,63 @@
+name: Static analysis
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'staging'
+  pull_request:
+  workflow_dispatch:
+
+env:
+  PYTEST_ADDOPTS: "--color=yes"
+
+jobs:
+  slither:
+    name: Slither run
+    runs-on: ubuntu-latest
+    defaults:
+        run:
+            shell: bash
+            working-directory: ./packages/contracts
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Fetch history
+        run: git fetch
+
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install Dependencies
+        # only install dependencies if there was a change in the deps
+        # if: steps.yarn-cache.outputs.cache-hit != 'true'
```

Comment:
Is this always reinstalling the dependencies?

Reply:
I believe so. 
TBH I'm mostly just copying what's [been](https://github.com/ethereum-optimism/optimism/blob/4c486d70b852d88eea014a5d1c432dafde1cbd4e/.github/workflows/static-analysis.yml#L41) [done](https://github.com/ethereum-optimism/optimism/blob/4c486d70b852d88eea014a5d1c432dafde1cbd4e/.github/workflows/ts-packages.yml#L41) elsewhere in this file.

Reply:
See https://github.com/actions/cache/blob/main/examples.md#node---yarn for examples.
I think we should enable a cache here (saves like half the time of CI)

Reply:
Hmm, hardhat doesn't seem to be in the `yarn.lock` file which is causing the issue. I'm not sure how it's installed.